TombEngine/Objects/TR2/Entity/tr2_sword_guardian.cpp

```diff
@@ -12,210 +13,372 @@
 #include "Game/misc.h"
 #include "Sound/sound.h"
 #include "Specific/level.h"
-
-using namespace TEN::Math::Random;
+#include "Specific/setup.h"
 
 namespace TEN::Entities::Creatures::TR2
 {
+	constexpr auto SWORD_GUARDIAN_FLY_TURN_RATE_MAX = ANGLE(7.0f);
+	constexpr auto SWORD_GUARDIAN_WALK_TURN_RATE_MAX = ANGLE(9.0f);
+
 	const auto SwordBite = BiteInfo(Vector3(0.0f, 37.0f, 550.0f), 15);
 
-	void SwordGuardianFly(ItemInfo* item)
+	constexpr auto SWORD_GUARDIAN_ATTACK_RANGE = SQUARE(SECTOR(1));
+	constexpr auto SWORD_GUARDIAN_WALK_ATTACK_RANGE = SQUARE(SECTOR(2));
+
+	constexpr auto SWORD_GUARDIAN_SWAPMESH_TIME = 3;
+	constexpr auto SWORD_GUARDIAN_ATTACK_DAMAGE = 300;
+
+	enum SwordGuardianState
+	{
+		SWORD_GUARDIAN_STATE_STOP = 1,
```

Comment:
Instead of **SWORD_GUARDIAN_STATE_STOP**
I think that **SWORD_GUARDIAN_STATE_IDLE** is more in line with the other entities states.

Reply:
Done